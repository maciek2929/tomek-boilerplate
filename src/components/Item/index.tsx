import React, { FC } from 'react';
import styles from './styles.module.scss';

type ItemProps = {
    addItem?: (name: string) => void;
    name: string;
};

const Item: FC<ItemProps> = ({ addItem, name }): JSX.Element => {
    
    const handleClick = (e: React.MouseEvent): void => {
        e.preventDefault();
        addItem && addItem(name)
    }

    return (
        <li onClick={handleClick} className={styles.item}>Name: {name}</li>
    );
}

export default Item;
