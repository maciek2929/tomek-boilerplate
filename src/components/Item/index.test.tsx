import React from "react";
import { shallow } from "enzyme";
import Item from "./index";

describe("Item", () => {
  it("renders", () => {
    const props = {
      addItem: () => { },
      name: "Test item name"
    };

    const item = shallow(<Item addItem={props.addItem} name={"asdw"} />);
    expect(item).toMatchSnapshot();
  });

  it("snapshot", () => {
    const props = {
      addItem: () => { },
      name: "Test item name"
    };

    const wrapper = shallow(<Item addItem={props.addItem} name={props.name} />);
    expect(wrapper.text()).toEqual("Name: Test item name")
  });

  it('handles click', () => {
    const mockCallBack = jest.fn();

    const button = shallow((<Item addItem={mockCallBack} name={"Product"} />));
    button.find('li').simulate('click', { preventDefault: () => { } });
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

});
