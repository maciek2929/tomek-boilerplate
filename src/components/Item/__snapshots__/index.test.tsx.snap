// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders 1`] = `ShallowWrapper {}`;
